### ISSUE TITLE
Keyboard shifting scrollview

### ISSUE THREAD

**Issue by TheRogue76** [FullStackCraft/react-native-keyboard-shift#5]
Right now, we need multiple packages just for Avoiding the Keyboard, Avoiding the keyboard on Android and [Avoiding the keyboard while in a Scroll view](https://github.com/wix/react-native-keyboard-aware-scrollview). Maybe create another component for the scrollview so that we can cover all use cases

**Comment by princefishthrower** [comment]
Yes - there is still another small optimization to make that when tapping outside of the keyboard area while the keyboard is open, the keyboard automatically closes - this should probably be a configurable prop like `shouldKeyboardAutoClose` - then we need to figure out how to actually implement that feature. An example of what I mean would be Whatsapp, where you can open a context menu on a message, and then the keyboard is finally dismissed. Currently, nothing outside the keyboard is focusable if you are scrolling and long pressing content whatever is above the keyboard.

Actually there is another thing too - the keyboard should be able to be swiped down to be dismissed, just like as in Whatsapp - I will open an issue for this.

**Comment by TheRogue76** [comment]
Auto closing the keyboard could be a very nice feature. i usually handle it with a view that detects onClick and dismisses the keyboard but it has never been as elegant as WhatsApp's dismiss feature. I'll work on something in a separate branch: ```feature/keyboardAutoClose```. We also need to agree on a branch strategy to avoid merge conflicts from now on. We could do gitflow but it usually wreaks havoc on CI/CD pipelines. Any Ideas?
